### PR TITLE
updated documentation for DeepHyper 0.2.5 on ThetaGPU

### PR DESCRIPTION
**Pull request by Deathn0t**
Hello @felker, 

I updated the documentation page of DeepHyper on this repo. However, I don't have the rights to create a custom module on ThetaGPU could someone follow the [new installation procedure](https://deephyper.readthedocs.io/en/latest/install/thetagpu.html#user-installation) and provide a DeepHyper module on ThetaGPU?

### CODE REVIEW

**Comment by felker** [comment]
@Deathn0t I am almost finished with the Lmod module for DeepHyper 0.2.5. It has:
- TensorFlow 2.4.1
- PyTorch `1.9.0a0+gitff982ef`
- Horovod 0.21.3

There are two problems with https://github.com/deephyper/deephyper/blob/master/setup.py namely https://github.com/deephyper/deephyper/blob/7ce685aba8ea2f8ce177aa66f5389245dcba84e5/setup.py#L77-L78

1. Typo with Horovod version? Should be 0.21.3, not 1.21.3
2. The Balsam version is relatively old, from Nov 2019. Has anyone tested with 0.5.0? Can we loosen the strict equality?

See https://github.com/deephyper/deephyper/pull/76

File: doc_staging/deephyper.md [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
 
-```bash
-source /lus/theta-fs0/software/thetagpu/balsam/setup.sh
-```
+The installation procedure of DeepHyper on ThetaGPU can be found on the official documentation of the software ["Installing DeepHyper on ThetaGPU](https://deephyper.readthedocs.io/en/latest/install/thetagpu.html).
+
+## Example
 
-From a ThetaGPU compute node, you may then initiate a local hyperparameter search using the `subprocess` evaluator.  For instance:
+From a ThetaGPU login node, you may then initiate an hyperparameter search using the `ray` evaluator through the `ray-submit` command line.  For instance:
```
Comment (felker) [inline doc_staging/deephyper.md]:
```suggestion
From a ThetaGPU service node (e.g. `thetagpusn1`), you may then initiate a hyperparameter search using the `ray` evaluator through the `ray-submit` command line.  For instance:
```

**Comment by felker** [comment]
Another possible problem: I have been installing DeepHyper with the following command in my module install script
```
pip install 'deephyper[analytics,balsam]==0.2.5'
```
But it seems that this doesnt automatically install `deepspace`? E.g. 
```
deephyper nas random --evaluator ray --ray-address auto --problem deephyper.benchmark.nas.mnist1D.problem.Problem --max-evals 10 --num-cpus-per-task 1 --num-gpus-per-task 1
```
fails because that dependency is missing

**Comment by Deathn0t** [comment]
`deepspace` is not yet installed by default. It is an extra `pip install deephyper[deepspace]`.

It is a new package that I created to provided a [library of accessible neural architecture search space](https://github.com/deephyper/deepspace)

**Pull request merged**